src/lib/adamant-api/index.js

```diff
@@ -129,6 +129,74 @@ export function getPublicKey(address = '') {
   })
 }
 
+
+/**
+ * @typedef {Object} MessageObject
+ * @property {number} amount
+ * @property {Object} asset
+ * @property {Object} asset.chat
+ * @property {string} asset.chat.message
+ * @property {string} asset.chat.own_message
+ * @property {number} asset.chat.type
+ * @property {string} recipientId
+ * @property {string} senderId
+ * @property {string} senderPublicKey
+ * @property {string} signature
+ * @property {number} timestamp
+ * @property {number} type
+ */
+
+/**
+ * Generates and signs a chat message transaction.
+ *
+ * @param {Object} params - The transaction parameters.
+ * @param {string} params.to - The recipient's identifier.
+ * @param {number} [params.amount=0] - The transaction amount.
+ * @param {number} [params.type=1] - The message type.
+ * @param {string|Object} params.message - The message to be encrypted.
+ * @returns {Promise<MessageObject|null>} The signed transaction object or null on failure.
+ */
+export async function signChatMessageTransaction(params) {
+  try {
+    const { to, amount, type = 1, message } = params;
+
+    const publicKey = await getPublicKey(to);
+
+    const text =
+      typeof message === 'string' ? message : JSON.stringify(message)
+    const encoded = utils.encodeMessage(text, publicKey, myKeypair.privateKey)
+    const chat = {
+      message: encoded.message,
+      own_message: encoded.nonce,
+      type,
+    }
+
+    const transaction = newTransaction(Transactions.CHAT_MESSAGE)
+    transaction.amount = amount ? utils.prepareAmount(amount) : 0
+    transaction.asset = { chat }
+    transaction.recipientId = to
+
+    const timeDelta = await client.getTimeDelta()
+
+    return signTransaction(transaction, timeDelta)
+  } catch (error) {
+    console.error(error)
+
+    return null
+  }
```

Comment:
Let it fail. We can catch that error in Vuex module. Fix the return type as well. It cannot return `null`.